src/viur/core/bones/uri.py

```diff
@@ -0,0 +1,204 @@
+            raise ValueError("domain_disallowed_list must be a list or a tuple or None")
+
+        if domain_allowed_list is not None:
+            if any([not isinstance(domain, str) for domain in domain_allowed_list]):
+                raise ValueError("domain_allowed_list must only contain strings")
+
+        if domain_disallowed_list is not None:
+            if any([not isinstance(domain, str) for domain in domain_disallowed_list]):
+                raise ValueError("domain_disallowed_list must only contain strings")
+
+        if domain_allowed_list and domain_disallowed_list:
+            raise ValueError("Only one of domain_allowed_list and domain_disallowed_list can be set")
+
+        if not isinstance(local_path_allowed, bool):
+            raise ValueError("local_path_allowed must be a boolean")
+
+        self.clean_get_params = clean_get_params
+        self.domain_allowed_list = domain_allowed_list
+        self.domain_disallowed_list = domain_disallowed_list
+        self.local_path_allowed = local_path_allowed
+
+    @classmethod
+    def _build_accepted_ports(cls, accepted_ports: str | int | t.Iterable[str | int]) -> list[range]:
+        if isinstance(accepted_ports, str):
+            if accepted_ports == "*":
+                return [range(PORT_MIN, PORT_MAX + 1)]
+
+            elif "," in accepted_ports:  # list of ranges, values
+                return cls._build_accepted_ports([
+                    value.strip() for value in accepted_ports.split(",")
+                ])
+
+            elif "-" in accepted_ports:  # range of ports
+                start, end = accepted_ports.split("-", 1)
+                start = int(start)
+                end = int(end)
+                if start > end:
+                    raise ValueError("Start value must be less than end value")
+
+                if start < PORT_MIN:
+                    raise ValueError("Start value must be greater than zero")
+
+                if end > PORT_MAX:
+                    raise ValueError(f"End value must be less or equal than {PORT_MAX}")
+
+                return [range(start, end + 1)]
+
+            else:
+                port = int(accepted_ports)
+                return [range(port, port + 1)]
+
+        elif isinstance(accepted_ports, int):
+            if accepted_ports < PORT_MIN:
+                raise ValueError("Port value must be greater than zero")
+
+            if accepted_ports > PORT_MAX:
+                raise ValueError(f"Port value must be less or equal than {PORT_MAX}")
+
+            return [range(accepted_ports, accepted_ports + 1)]
+
+        elif isinstance(accepted_ports, Iterable):
+            accepted_ports_value = []
+            for accepted_port in accepted_ports:
+                accepted_ports_value.extend(UriBone._build_accepted_ports(accepted_port))
+            return accepted_ports_value
+
+        raise ValueError("accepted_ports must be a iterable or an integer or string")
```

Comment:
`accepted_ports` defaults to `None`, so when not given, the instanciation fails with 
```
  File "~/myproject/deploy/skeletons/test.py", line 5, in <module>
    class TestSkel(skeleton.Skeleton):
  File "~/myproject/deploy/skeletons/test.py", line 17, in TestSkel
    url = UriBone(
          ^^^^^^^^
  File "~/mycore/src/viur/core/bones/uri.py", line 45, in __init__
    self.accepted_ports = set(sorted(UriBone._build_accepted_ports(accepted_ports), key=lambda rng: rng.start))
                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/mycore/src/viur/core/bones/uri.py", line 130, in _build_accepted_ports
    raise ValueError("accepted_ports must be a iterable or an integer or string")
ValueError: accepted_ports must be a iterable or an integer or string
```